lib/msf/ui/console/command_dispatcher/auxiliary.rb

```diff
@@ -79,21 +79,29 @@ def cmd_run(*args)
     jobify  = false
     quiet   = false
 
-    @@auxiliary_opts.parse(args) { |opt, idx, val|
+    @@auxiliary_opts.parse(args) do |opt, idx, val|
       case opt
-        when '-j'
-          jobify = true
-        when '-o'
-          opt_str = val
-        when '-a'
-          action = val
-        when '-q'
-          quiet  = true
-        when '-h'
+      when '-j'
+        jobify = true
+      when '-o'
+        opt_str = val
+      when '-a'
+        action = val
+      when '-q'
+        quiet  = true
+      when '-h'
+        cmd_run_help
+        return false
+      else
+        (key, val) = val.split('=')
```

Comment:
So I'm curious how this can be consolidated with the `-o` option above. There are two techniques being employed here - both work. `-o` sets `OptionStr` in the module run, where `Msf::DataStore#import_options_from_s` eventually runs on it.

Reply:
good point. -o is obsolete more-or-less, so we could just make it a no-op and it would work exactly the same as before.

Reply:
good question. to be honest, I was not even aware that this existed until you pointed it out!

Reply:
This was finally resolved in #12198.